### ISSUE TITLE
[Molten Core - Mage] Iceblock shouldn't reset boss

### ISSUE THREAD

**Issue by Elexias** [chromiecraft/chromiecraft#2752]
### What client do you play on?

enUS

### Faction

- [ ] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [X] 60

### Current Behaviour

![image](https://user-images.githubusercontent.com/80590226/148661123-a1df6ebb-b9c4-4359-b6a4-ec0b4bde1180.png)

We did alt run, and I was on mage, we wiped on 1% hp and hunter used feign dead, I used Iceblock and fight did reset, Iceblock shouldn't reset boss it is not Invisibility.

### Expected Blizzlike Behaviour

Iceblock shouldn't reset boss it is not Invisibility.

### Steps to reproduce the problem

1. Form raid group and make sure that you're last alive or have hunter alive who can FD
2. Let him FD
3. Use Iceblock and fight is reset

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/86f1b87fe60f462a8045ee169b72fe1104140100

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by claudiodfc** [comment]
Provide source.

**Comment by Elexias** [comment]
Because every time you use iceblock out of world or anywhere mobs keeps attacking you until you get out of it?

**Comment by ridai260** [comment]
Yes you should still be able to be attacked and remain in combat even when using Ice Block against any mob. This is how it's always been on vanilla WoW. It should NOT remove you from combat, if the above report is true then it currently seems to be acting the same way as Feign Death which is wrong (on ability use dropping aggro). 

I wonder if this is also true in PvP or only PvE (i.e. if you are fighting a mage that ice blocks do they leave PvP combat?).

**Comment by Nyeriah** [comment]
It's an invulnerability like paladin's divine shield

**Comment by Clusterfonk** [comment]
Tested it on my local clean server and I can reproduce it on every boss in molten core. 
Also works when the mage Iceblocks first and then the hunter FDs.

Does not work in openworld or in other dungeons/raids (tried Zul Gurub and Mara)

**Comment by ishidinho** [comment]
Hey @Elexias and all the other guys. Many thanks for your work on the report.

I can confirm the issue for Iceblock and Divine Shield.
